mdbook-course/src/markdown.rs

```diff
@@ -57,6 +58,46 @@ pub fn duration(mut minutes: u64) -> String {
     }
 }
 
+/// Table implements Display to format a two-dimensional table as markdown,
+/// following https://github.github.com/gfm/#tables-extension-.
+pub struct Table<const N: usize> {
```

Comment:
Nice! If you have time, it would be nice to have a small unit test below to show how the output looks like.